apps/gitness/src/pages/repo-sandbox/repo-sandbox-list.tsx

```diff
@@ -26,22 +26,45 @@ const LinkComponent = ({ to, children }: { to: string; children: React.ReactNode
 
 export default function ReposSandboxListPage() {
   const space = useGetSpaceURLParam()
+  const navigate = useNavigate()
 
   /* Query and Pagination */
   const { query: currentQuery = '', sort } = useCommonFilter<ListReposQueryQueryParams['sort']>()
   const [query, _] = useQueryState('query', { defaultValue: currentQuery })
   const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
 
-  const { isFetching, data: { body: repositories, headers } = {} } = useListReposQuery({
-    queryParams: { sort, query, page },
-    space_ref: `${space}/+`
-  })
+  const {
+    isFetching,
+    data: { body: repositories, headers } = {},
+    isError,
+    error
+  } = useListReposQuery(
+    {
+      queryParams: { sort, query, page },
+      space_ref: `${space}/+`
+    },
+    { retry: false }
+  )
 
   const totalPages = parseInt(headers?.get(PageResponseHeader.xTotalPages) || '')
 
   const renderListContent = () => {
     if (isFetching) return <SkeletonList />
 
+    if (isError)
+      return (
+        <NoData
+          title="Error"
+          description={[error.message || '']}
```

Comment:
NIT: Maybe we could provide a default error message in case `error.message` is null/undefined?

Reply:
Good idea. Will add in a follow up PR.